VIC_RBM/model_run/result_analysis/scripts/plot_flow_future.py

```diff
@@ -0,0 +1,129 @@
+
+hist_route_path = args.hist_route_path # historical run. Lohmann routing output. year; month; day; flow(cfs)
+future_route_paths = args.future_route_paths # future run. Lohmann routing output. year; month; day; flow(cfs); ['path1', 'path2']
+future_route_names = args.future_route_names # future run. year; month; day; flow(cfs); ['Scenario1', 'Scenario2']
+
+output_plot_basename = '../output/Tennessee_' + usgs_site_code # output plot path basename (suffix will be added to different plots)
+
+#-------------------------------------------------
+
+plot_hist_start_date = dt.datetime(1950, 10, 1)  # start date shown on the plot for historical run (should be complete water years)
+plot_hist_end_date = dt.datetime(1999, 9, 30)  # end date shown on the plot for historical run
+
+plot_future_start_date = dt.datetime(1950, 10, 1)  # start date shown on the plot for future run (should be complete water years)
+plot_future_end_date = dt.datetime(2099, 9, 30)  # end date shown on the plot for future run
+
+time_locator = ('year', 20)  # time locator on the plot; 'year' for year; 'month' for month. e.g., ('month', 3) for plot one tick every 3 months
+
+#-------------------------------------------------
+
+nfuture = len(future_route_paths)  # number of future scenarios
+
+hist_style = 'k-'
+style_options = ['b--', 'r--', 'g--', 'm--', 'y--']
+list_style_future = []  # a list of plotting styles of future scenarios
+for i in range(nfuture):
+	list_style_future.append(style_options[i])
+
+#========================================================
+# Load data
+#========================================================
+s_hist = my_functions.read_Lohmann_route_daily_output(hist_route_path)
+list_s_future = []  # a list contains pd.Series of data of each futre scenario
+for i in range(nfuture):
+	s_future = my_functions.read_Lohmann_route_daily_output(future_route_paths[i])
+	list_s_future.append(s_future)
+
+#========================================================
+# Select data to be plotted
+#========================================================
+s_hist_to_plot = my_functions.select_time_range(s_hist, plot_hist_start_date, plot_hist_end_date)
+list_s_future_to_plot = []
+for i in range(nfuture):
+	s_future_to_plot = my_functions.select_time_range(list_s_future[i], plot_future_start_date, plot_future_end_date)
+	list_s_future_to_plot.append(s_future_to_plot)
+
+#========================================================
+# plot
+#========================================================
+model_info = 'VIC runoff, historical -  1/8 deg, Maurer forcing, Andy Wood setup\n          VIC runoff, climate change scenarios - 1/8 deg, Reclamation archive (GCM forcing)\n          Route: Lohmann, Wu flowdir, Andy Wood setup'
+
+##============== plot daily data ===============#
+#fig = my_functions.plot_time_series(plot_date=True, \
+#		list_s_data=[s_hist_to_plot]+list_s_future_to_plot, \
+#		list_style=[hist_style]+list_style_future, \
```

Comment:
The list_s_data, list_style, and list_label items could all be included into one dictionary which could make it more user-friendly. And then you could even make a dictionary with all of your plotting stylistic options and pass it to matplotlib as **kwargs (this is something I've been trying to do but haven't gotten to work yet). Like in this link: http://stackoverflow.com/questions/334655/passing-a-dictionary-to-a-function-in-python-as-keyword-parameters